<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="6" total="12" passed="4" failed="2" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2025-08-20T08:48:46 IST" name="DemoTestNGSuite" finished-at="2025-08-20T08:48:46 IST" duration-ms="461">
    <groups>
      <group name="expectedexceptions">
        <method signature="Testnglearn.testcase5()[pri:0, instance:Testcases.Testnglearn@10a9d961]" name="testcase5" class="Testcases.Testnglearn"/>
        <method signature="Testnglearn.testcase6()[pri:0, instance:Testcases.Testnglearn@10a9d961]" name="testcase6" class="Testcases.Testnglearn"/>
        <method signature="Testnglearn.testcase4()[pri:0, instance:Testcases.Testnglearn@10a9d961]" name="testcase4" class="Testcases.Testnglearn"/>
        <method signature="Testnglearn.testcase5()[pri:0, instance:Testcases.Testnglearn@554e218]" name="testcase5" class="Testcases.Testnglearn"/>
        <method signature="Testnglearn.testcase6()[pri:0, instance:Testcases.Testnglearn@554e218]" name="testcase6" class="Testcases.Testnglearn"/>
        <method signature="Testnglearn.testcase4()[pri:0, instance:Testcases.Testnglearn@554e218]" name="testcase4" class="Testcases.Testnglearn"/>
      </group> <!-- expectedexceptions -->
    </groups>
    <test started-at="2025-08-20T08:48:46 IST" name="TestNG Testing practices" finished-at="2025-08-20T08:48:46 IST" duration-ms="298">
      <class name="Testcases.Testnglearn">
        <test-method signature="testcase4()[pri:0, instance:Testcases.Testnglearn@10a9d961]" started-at="2025-08-20T08:48:46 IST" name="testcase4" finished-at="2025-08-20T08:48:46 IST" duration-ms="40" status="PASS">
          <exception class="java.lang.ArithmeticException">
            <message>
              <![CDATA[/ by zero]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.ArithmeticException: / by zero
at pages.Testnglearnings.checkException(Testnglearnings.java:13)
at Testcases.Testnglearn.testcase4(Testnglearn.java:65)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:840)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.ArithmeticException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testcase4 -->
        <test-method signature="testcase5()[pri:0, instance:Testcases.Testnglearn@10a9d961]" started-at="2025-08-20T08:48:46 IST" name="testcase5" finished-at="2025-08-20T08:48:46 IST" duration-ms="40" status="FAIL">
          <exception class="java.lang.ArithmeticException">
            <message>
              <![CDATA[/ by zero]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.ArithmeticException: / by zero
at pages.Testnglearnings.checkException(Testnglearnings.java:13)
at Testcases.Testnglearn.testcase5(Testnglearn.java:69)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:840)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.ArithmeticException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testcase5 -->
        <test-method signature="testcase6()[pri:0, instance:Testcases.Testnglearn@10a9d961]" started-at="2025-08-20T08:48:46 IST" name="testcase6" finished-at="2025-08-20T08:48:46 IST" depends-on-methods="Testcases.Testnglearn.testcase5" duration-ms="6" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testcase6 -->
      </class> <!-- Testcases.Testnglearn -->
    </test> <!-- TestNG Testing practices -->
    <test started-at="2025-08-20T08:48:46 IST" name="TestNG running in safari" finished-at="2025-08-20T08:48:46 IST" duration-ms="39">
      <class name="Testcases.Testnglearn">
        <test-method signature="testcase5()[pri:0, instance:Testcases.Testnglearn@554e218]" started-at="2025-08-20T08:48:46 IST" name="testcase5" finished-at="2025-08-20T08:48:46 IST" duration-ms="3" status="FAIL">
          <exception class="java.lang.ArithmeticException">
            <message>
              <![CDATA[/ by zero]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.ArithmeticException: / by zero
at pages.Testnglearnings.checkException(Testnglearnings.java:13)
at Testcases.Testnglearn.testcase5(Testnglearn.java:69)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:840)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.ArithmeticException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testcase5 -->
        <test-method signature="testcase4()[pri:0, instance:Testcases.Testnglearn@554e218]" started-at="2025-08-20T08:48:46 IST" name="testcase4" finished-at="2025-08-20T08:48:46 IST" duration-ms="6" status="PASS">
          <exception class="java.lang.ArithmeticException">
            <message>
              <![CDATA[/ by zero]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.ArithmeticException: / by zero
at pages.Testnglearnings.checkException(Testnglearnings.java:13)
at Testcases.Testnglearn.testcase4(Testnglearn.java:65)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:840)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.ArithmeticException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testcase4 -->
        <test-method signature="testcase6()[pri:0, instance:Testcases.Testnglearn@554e218]" started-at="2025-08-20T08:48:46 IST" name="testcase6" finished-at="2025-08-20T08:48:46 IST" depends-on-methods="Testcases.Testnglearn.testcase5" duration-ms="9" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testcase6 -->
      </class> <!-- Testcases.Testnglearn -->
    </test> <!-- TestNG running in safari -->
  </suite> <!-- DemoTestNGSuite -->
</testng-results>
